pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -730,6 +744,8 @@ func (s *DiscoveryServer) updateProxyServiceInstances(instance *model.ServiceIns
 		return nil
 	}
 
+	proxy.ServiceInstances = instances
```

Comment:
why can this prevent data race? This is still in diff goroutines with pushConnection.

Reply:
This is now being called from the same goroutine: both `pushConnection()` and `doUpdateProxyServiceInstances()` are called from `StreamAggregatedResources()`.

Reply:
I remembered